src/artifact-uploader.cpp

```diff
@@ -19,28 +27,49 @@ namespace connect
 class ArtifactUploader::Impl
 {
 public:
+    Impl()
+        : accountId_(-1)
+        , cameraId_(-1)
+        , done_(false)
+        , timeoutToCompleteUploadSec_(0)
+    {
+    }
+
     ~Impl();
 
     Status init(const ArtifactUploader::Configuration& cfg,
                 ArtifactUploader::ClientConfigCallback* configCallback);
 
-    // all upload* methods are asynchronous, non-blocking, take ownership
-    // of data passed to them
-    Status uploadBackground(const timestamp_t& timestamp, PayloadHolderPtr payload);
-    Status uploadObjectStream(const ObjectStream& stream, PayloadHolderPtr payload);
-    Status uploadFlipbook(const Flipbook& flipbook, PayloadHolderPtr payload);
-    Status uploadEvent(const timestamp_t& timestamp, move_ref<Events> data);
-    Status uploadCount(move_ref<Counts> counts);
+    Status enqueueTask(UploadArtifactTaskPtr task)
+    {
+        return queue_->push_back(task);
+    }
+
+    void abort()
+    {
+        done_ = true;
+    }
 
 private:
-    OutputControllerPtr outputController_;
```

Comment:
Controller was inlined, so 99% of its logics is now in ArtifactUploader::Impl::threadFunc() andn ArtifactUploader::Impl::~Impl().
Minus one class and two files, and dozen methods.